### DISCUSSION TITLE
Welcome to pixel-gantry-control Discussions!

### DISCUSSION THREAD

**Discussion by cfangmeier** [CUASAS/pixel-gantry-control#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nnd26-hub** [comment]
Hello, I just installed the latest release of gScript on a PC connected to the gantry. The gScript Interpreter UI appears as expected. However, when pressing the run arrow, the following error message appears. I would really appreciate support in debugging this issue. 
![IMG_8985](https://github.com/user-attachments/assets/502c178d-2f8b-4fe3-a22a-28f2c01203af)

**Reply by cfangmeier** [reply]
Greetings @nnd26-hub . A few questions that could help point towards what is going on: 

  - Which version of gScript did you download? A3200 or Automation1?
  - Do you have the matching Aerotech software installed?
  - What version of Windows are you using?
  - Have you been able to run any previous versions of gScript? If so, which ones and what else might have changed since they were working?

**Comment by jmonroym** [comment]
Hi Caleb, This is Jose. 

We had an unfortunate issue here with our PC and we needed to reimage the HD to win 10. That is what we have currently. We reinstalled Labview 2025 Q3.  All packages are 2025 Q3 version.  

We reinstalled A3200 (6.04.004).

We have Noor, a new undergrad student working in the gscript setup. I think she downloaded the A3200 version 2.8.6

**Reply by cfangmeier** [reply]
Hi Jose, Typically this kind of error is due to some missing library or mismatched library versions. Of course, LabVIEW is not helpful enough to say which libraries in particular it is having trouble with. You say you have installed A3200 which is a great start. I believe any iteration of version 6 should work, although the .dll versions being distributed with gScript are 6.4.

The other required dependency is IDS peak. I suggest trying to follow the installation instructions [here](https://github.com/CUASAS/pixel-gantry-control/releases/tag/v2.8.0) and see how that goes. If you still cannot launch gScript, let me know and we can investigate further.

**Comment by jmonroym** [comment]
I think the issue is with one particular library. When launching the gscript application, labview complains about a missing library called "lvwebclient_64.dll". as shown in the screenshot  below. I already installed IDS peak, so I will test the sytem again and bet back to you. Thanks  
<img width="433" height="235" alt="1" src="https://github.com/user-attachments/assets/597d2d76-4573-40e1-b1b7-988eef5e0f42" />

**Reply by cfangmeier** [reply]
Hmm. That's interesting. I double-checked and that dll *should* be included in the installer. Can you check in the installation folder `C:\Program Files\gScript\gScript - A3200\data` to see if it's there?

**Reply by cfangmeier** [reply]
And if it is there, what happens if you navigate to and select it during startup?

**Reply by jmonroym** [reply]
I found it. The application was not installed in the right place but in the desktop so the libraries and all the files are over there.  The application ran normally once I linked the library. I did not try functionality yet. I will probably reinstall the application to make sure files are in the correct place. 
Once before you suggested me to use the previous to the latest version of the application to make sure all the bugs are fixed; is it still a good advice?
Thanks Caleb.

**Reply by cfangmeier** [reply]
Good to hear that you found the issue. Several months ago, there were some issues with the A3200 implementation, but but as far as I'm aware things are now working. So, I recommend trying the most recent version and let me know if you run into any problems.

**Comment by jmonroym** [comment]
<img width="372" height="272" alt="Capture" src="https://github.com/user-attachments/assets/cd0fcc62-e835-4da3-b2fe-6147bb0d04d2" />

Hello Caleb, 

The application has been successfully reinstalled in the correct location. After selecting virtual mode, this error message appeared. What is your advice on proceeding from here.

**Reply by cfangmeier** [reply]
The error indicates that there is an issue with the entries in your flex_config related to the camera setup. Can you attach your flex_config so I can take a look at it. Alternatively, you can check out the config instructions [here](https://github.com/CUASAS/pixel-gantry-control/releases/tag/v2.8.0). Were you able to get IDS peak working?

**Comment by nnd26-hub** [comment]
Hi Caleb, 
<img width="1579" height="330" alt="image" src="https://github.com/user-attachments/assets/a9cd3a7c-0037-488a-a708-f697706afb02" />



This error was encountered even when trying to run [camera.vi](http://camera.vi/), and for some reason the camera inside NI MAX is no longer accessible (critical error).

This error may have arose after running the camera and peak finding and having the peak finding fail mid -way, then exiting. But I'm not sure.

Thanks
Timo and Noor

**Comment by cfangmeier** [comment]
Hi Timo and Noor,

Since the camera was previously working, probably it just somehow ended up in a bad state. I suggest doing the following to reset it:

  1. Close gScript entirely
  2. Unplug and replug the camera
  3. See if you can get a video feed using the "IDS peak Cockpit" application
  4. If that's good, close it, re-launch gScript. If your config is correct it should work.
  5. If you can't get a video feed in peak Cockpit, try restarting the PC.
  6. If that still doesn't work, follow-up here and we can try digging deeper into the issue.

**Comment by strombolini** [comment]
Caleb,
in gscript one of my programs uses CMD_MOVEREL.vi which used to load dependencies from a folder called Aerotech Automation1. But this folder I can't find anymore. Specifically, I searched the whole computer for Task Stop.vi and couldn't find anything. Do you know if reinstalling gscript would fix this or can I find these VI's somewhere on the web?

Thank you for your help

Timo

**Reply by cfangmeier** [reply]
Hi Timo,

The Automation1 VIs come with the Aerotech software. Specifically, you want to have installed the Automation1 MDK. After installing it, you also need to run "C:\Program Files\Aerotech\Automation1-MDK\Bin\Automation1LabVIEWInstaller.exe" to unpack and install install the LabVIEW API. 

-Caleb

**Reply by strombolini** [reply]
Hi Caleb, we unfortunately don't have Automation1 according to Jose, who said we don't have compatible hardware. Any advice here? Are there any alternatives to Automation1 or should we try to somehow brute force the installation?

Thanks 
Timo

**Reply by cfangmeier** [reply]
I see. In that case you just need the VIs themselves so labview is happy. Of course you will not be able to actually execute the VIs, but since you are not using Automation1 anyway, that shouldn't be a problem. I've attached a zip file with the VIs. Just unpack them to somewhere convenient and point labview to them when you load the project.

[Aerotech Automation1.zip](https://github.com/user-attachments/files/23940881/Aerotech.Automation1.zip)

**Comment by jmonroym** [comment]
Hi all, Just a question: as far as I know Gscript  can generate .txt output files, for final reports for instance; is there a way to generate files in other formats, like .csv? Of course I can always use a python script to convert .txt to .csv, however I am wondering if this is a functionality that might be useful as a native feature. In our case, we are making a survey of the final module position on the DEE, using the gantry camera and gscript. I would lije to have as output a .csv file with the module's corners coordinates and fitting results.
best

**Reply by cfangmeier** [reply]
Hey Jose.

There isn't any built-in system for exporting CSV, but I think you can still achieve what you want with a series of print commands. Something like the following should work as a starting point:

```bash
SETLOG output_file.csv # set to wherever you want the output to be, separated from your normal log
CLEARLOG # omit if you want to append to the log file, rather than overwrite
# use XPRINT to omit the timestamp prefix
XPRINT "MOD_1_TR,{$mod_1_tr.x:%.3f},{$mod_1_tr.y:%.3f},{$mod_1_tr.z:%.3f}"
XPRINT "MOD_1_TL,{$mod_1_tl.x:%.3f},{$mod_1_tl.y:%.3f},{$mod_1_tl.z:%.3f}"
XPRINT "MOD_1_BR,{$mod_1_br.x:%.3f},{$mod_1_br.y:%.3f},{$mod_1_br.z:%.3f}"
XPRINT "MOD_1_BL,{$mod_1_bl.x:%.3f},{$mod_1_bl.y:%.3f},{$mod_1_bl.z:%.3f}"

# ... repeat for other modules

SETLOG regular_log.txt # set the log back to your normal log output
```

Note that this uses the improved string formatting added in version 2.9.0.

**Reply by jmonroym** [reply]
great, we will try !
